pkg/storage/enhanced/azblob.go

```diff
@@ -0,0 +1,319 @@
+
+	// Use parallel workers since the old Azure SDK doesn't support batch operations
+	return az.deleteParallel(ctx, keys)
+}
+
+// deleteParallel uses parallel workers for delete operations
+func (az *EnhancedAzureBlob) deleteParallel(ctx context.Context, keys []string) (*BatchResult, error) {
+	log.Debug().Int("key_count", len(keys)).Msg("using Azure Blob parallel delete")
+
+	workerCount := az.getOptimalWorkerCount(len(keys))
+	workerPool := az.createWorkerPool(workerCount)
+
+	// Start workers
+	workerPool.start(ctx)
+	defer workerPool.stop()
+
+	// Send jobs
+	go func() {
+		defer close(workerPool.jobs)
+		for _, key := range keys {
+			job := AzureBlobDeleteJob{
+				blobName: key,
+				result:   workerPool.results,
+			}
+			select {
+			case workerPool.jobs <- job:
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	// Collect results
+	var successCount int
+	var failedKeys []FailedKey
+	var errors []error
+
+	for i := 0; i < len(keys); i++ {
+		select {
+		case result := <-workerPool.results:
+			if result.error != nil {
+				failedKeys = append(failedKeys, FailedKey{
+					Key:   result.blobName,
+					Error: result.error,
+				})
+				errors = append(errors, fmt.Errorf("failed to delete %s: %w", result.blobName, result.error))
+			} else {
+				successCount++
+			}
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+
+	az.metrics.FilesDeleted = int64(successCount)
+	az.metrics.FilesFailed = int64(len(failedKeys))
+
+	return &BatchResult{
+		SuccessCount: successCount,
+		FailedKeys:   failedKeys,
+		Errors:       errors,
+	}, nil
+}
+
+// createWorkerPool creates a new worker pool for parallel operations
+func (az *EnhancedAzureBlob) createWorkerPool(workerCount int) *AzureBlobWorkerPool {
+	return &AzureBlobWorkerPool{
+		container: az.container,
+		workers:   workerCount,
+		jobs:      make(chan AzureBlobDeleteJob, workerCount*2),
+		results:   make(chan AzureBlobDeleteResult, workerCount*2),
+	}
+}
+
+// getOptimalWorkerCount determines the optimal number of workers
+func (az *EnhancedAzureBlob) getOptimalWorkerCount(jobCount int) int {
+	// Start with default max workers for Azure
+	maxWorkers := 20 // Conservative default for Azure Blob
+
+	// Don't create more workers than jobs
+	if jobCount < maxWorkers {
+		maxWorkers = jobCount
+	}
+
+	// Ensure we have at least 1 worker
+	if maxWorkers < 1 {
+		maxWorkers = 1
+	}
+
+	return maxWorkers
+}
+
+// updateThroughputMetrics calculates and updates throughput metrics
+func (az *EnhancedAzureBlob) updateThroughputMetrics() {
+	if az.metrics.TotalDuration > 0 && az.metrics.BytesDeleted > 0 {
+		throughputBytes := float64(az.metrics.BytesDeleted) / az.metrics.TotalDuration.Seconds()
+		az.metrics.ThroughputMBps = throughputBytes / (1024 * 1024) // Convert to MB/s
+	}
+}
+
+// SupportsBatchDelete returns false as the old Azure SDK doesn't support batch operations
+func (az *EnhancedAzureBlob) SupportsBatchDelete() bool {
+	return false // Old SDK doesn't support batch delete
```

Comment:
why we generate this code in this case?

maybe better refactoring azblob to use modern SDK first?